core/src/main/java/org/infinispan/transaction/impl/TransactionTable.java

```diff
@@ -471,10 +473,17 @@ public void remoteTransactionCommitted(GlobalTransaction gtx, boolean onePc) {
    }
 
    public final RemoteTransaction removeRemoteTransaction(GlobalTransaction txId) {
-      RemoteTransaction removed = remoteTransactions.remove(txId);
-      if (trace) log.tracef("Removed remote transaction %s ? %s", txId, removed);
-      releaseResources(removed);
-      return removed;
+      ByRef<RemoteTransaction> removed = new ByRef<>(null);
+      // we need to mark the transaction inside compute() even if it does not exist
```

Comment:
Why not mark the tx as completed before removing it from the map?

Reply:
It seems I could; I thought I have an counterexample but probably I was mistaken.

Reply:
Actually I need to see if the tx was `successful`; trying to mark before remove/compute means that I'd need to do another lookup into `remoteTransaction` anyway.

Reply:
I missed that, let's keep it in `compute()` then.